lib/model/pet.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter/cupertino.dart';
+
+class Pet {
+  String uid;
+  String ownerId;
+  String name;
+  String type;
+  String biography;
+  String pictureUrl;
+  int age;
+  bool forPetSitting;
+  bool forPetMating;
+
+  Pet(this.uid, this.ownerId, this.name, this.type, this.age)
+      : biography = "",
+        pictureUrl = "",
+        forPetSitting = false,
+        forPetMating = false;
+
+  Pet.complete(this.uid, this.ownerId, this.name, this.type, this.biography,
+      this.pictureUrl, this.age, this.forPetMating, this.forPetSitting);
+
+  Pet.keyVal(
+      {@required this.uid,
+      @required this.ownerId,
+      @required this.name,
+      @required this.type,
+      @required this.age})
+      : biography = "",
+        pictureUrl = "",
+        forPetSitting = false,
+        forPetMating = false;
+
+  Pet.keyValComplete(
+      {@required this.uid,
+      @required this.ownerId,
+      @required this.name,
+      @required this.type,
+      @required this.biography,
+      @required this.pictureUrl,
+      @required this.age,
+      @required this.forPetSitting,
+      @required this.forPetMating});
+
+  Pet.empty()
+      : uid = "",
+        ownerId = "",
+        name = "",
+        type = "",
+        biography = "",
+        pictureUrl = "",
+        age = 0,
+        forPetSitting = false,
+        forPetMating = false;
+
+  Map<String, dynamic> toJson() => {
+        'uid': uid,
```

Comment:
Let's not save this field in the database object as it can't be initialized on creation.
When we will implement the `getPet` method, we can set the pet's `id` in-memory based on the `DocumentReference`'s id.